### ISSUE TITLE
Improve focus management in the KB search results

### ISSUE THREAD

**Issue by afercia** [Yoast/yoast-components#338]
WHen using a keyboard and activating one of hte KB search results links, the component gets re-rendered:
- the currently focused element (the link) gets removed from the DOM
- focus is lost

In the same way, when in a search detail view, activating the "Go back" button produces a focus loss.

To improve keyboard navigation, fosu should be programmatically set on a proper element.

To evaluate: set focus on the tabpanel itself, which is already focusable and labeled with `aria-labelledby`. Unfortunately, it's part of another component.

**Comment by afercia** [comment]
Struggling a bit with this:
- there's no global state to use
- `ref` doesn't work in react-tabs 🙁  see https://github.com/reactjs/react-tabs/issues/203
- even if it worked, we should then make the children aware of the parent refs

**Comment by afercia** [comment]
Refs don't work with styled components 😬  https://github.com/styled-components/styled-components/issues/618#issuecomment-289247967

**Comment by afercia** [comment]
Seems to me it is not even possible to get the clicked result link with `event.target` and store it somewhere, because when going back from Detail to Search, the component gets re-rendered and the stored DOM node is not the same any longer.
